### PR TITLE
Milestone5

### PR DESCRIPTION
**Pull request by gonjavi**
This pull request:

App component
- The App component is going to be your stateful parent component. Initialize its state in the constructor with three properties: total, next and operation, just like in your calculate.js module. By default, they should all be set to null.
Implement the handleClick function which takes buttonName as an argument and changes the state based on the result of the calculate function.
Pass the total or next value to Display to display the current result.
Pass handleClick as a clickHandler property to ButtonPanel
- ButtonPanel
Implement the handleClick function which takes buttonName as an argument and returns clickHandler from props.
Pass the clickHandler property to each button in the panel.
- Button
Implement the handleClick function which takes buttonName as an argument and returns clickHandler from props.
Use the onClick event to attach the handleClick event to the button.

### CODE REVIEW

**Comment by gonjavi** [comment]
Trying to see what the problems are

File: src/components/App.js [hunk 1]
```diff
@@ -2,14 +2,32 @@ import React from 'react';
...
+    const data = Calculate(this.state, buttonName);
+    this.setState(data);
+  }
+
+  render() {
+    const { total, next } = this.state;
+    return (
+      <div className="App">
+        <Display result={total || next} />
+        <ButtonPanel onClick={this.handleClick} />
```
Comment (azdrenymeri) [inline src/components/App.js:27]: Your `handleClick` is declared with the uppercase `H` on line 17 of this file so make sure to make the function downcase.
That's the reason for the error every time you click any of the buttons of the calculator.

File: src/components/App.js [hunk 2]
```diff
@@ -2,14 +2,32 @@ import React from 'react';
...
+    super(props);
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+  }
+
+  HandleClick(buttonName) {
+    const data = Calculate(this.state, buttonName);
```
Comment (azdrenymeri) [inline src/components/App.js:18]: Calculate is a function not a class change it to a lower case here and on the calculate.js file.
From `Calculate` to `calculate`

**Comment by gonjavi** [comment]
Thanks,
I changed the requested, and I tried to find what the error was, but no result.
I'm going to request another review to see what my problem is because I don't see it

File: src/components/Button.js [hunk 3]
```diff
@@ -51,7 +52,7 @@ const Button = props => {
     };
   }
   return (
-    <div style={styleButton}>{buttonName}</div>
+    <button style={styleButton} onClick={() => onClick(buttonName)} type="button">{buttonName}</button>
```
Comment (bellom) [inline src/components/Button.js:55]: I don't seem to know where `onClick` function came from, I can't see where you declared it.

File: src/components/ButtonPanel.js [hunk 4]
```diff
@@ -8,59 +9,45 @@ const styleGroup1 = {
...
-    <Button buttonName="." color="#dfe8f7" />
-    <Button buttonName="=" rightB="yes" />
-  </div>
-);
+const ButtonPanel = props => {
+  const { onClick } = props;
+  return (
+    <div className="style">
+      <div className="styleGroup1" style={styleGroup1}>
+        <Button buttonName="AC" onClick={onClick} color="#dfe8f7" />
```
Comment (bellom) [inline src/components/ButtonPanel.js:17]: this is `<ButtonPanel onClick={this.handleClick} />` what you have in `app.js`

so you should pass the props to the bind function created in `app.js` 
example:
`<Button buttonName="AC" handleClick = { onClick } color="#dfe8f7" />`

File: src/components/Button.js (hunk 3 above)
Comment (gonjavi) [reply src/components/Button.js:55]: thanks

File: src/components/ButtonPanel.js (hunk 4 above)
Comment (gonjavi) [reply src/components/ButtonPanel.js:17]: thanks

**Pull request merged**